instrumentation/rpc/src/test/java/brave/rpc/ITRpcTracingClassLoader.java

```diff
@@ -20,7 +20,9 @@
 import static brave.test.util.ClassLoaders.assertRunIsUnloadable;
 import static org.assertj.core.api.Assertions.assertThat;
 
-public class RpcTracingClassLoaderTest {
+// Mockito taints the classloader, through its log initialization. While this isn't an IT, this
+// works around the problem by forcing a different classloader.
+public class ITRpcTracingClassLoader {
```

Comment:
I wonder if instead forcing the mockito using test classes into their own class loader would be a nicer work around.

This person has a solution to force the mockito classes into their own loader, maybe that could be extended to the things that they load https://stackoverflow.com/a/28657203/2232476

Reply:
I'll change the comment. the problem is that it is initializes logging. Our current classpath test thing cannot unload the log manager.

Reply:
I'll leave a note about your comment, as I don't have unlimited time to dig into these things and what I did is a pragmatic workaround. Someone later can run into the rabbit hole, and hopefully will, but there's really no problem with how it is working now.